openedx/adg/admin_dashboard/applications/views.py

```diff
@@ -1,3 +1,202 @@
+        return self.filter_and_sort_applications(applications)
+
+    def get_paginate_by(self, queryset):
+        """
+        Allows dynamic pagination as selected by the user
+        """
+        self.paginate_by = self.request.GET.get('paginate-by', self.paginate_by)
+        return self.paginate_by
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        """
+        Prepare and return the extra context required by the listing pages i.e query params,
+        application type, total number of applications etc
+        """
+        context = super().get_context_data(object_list=object_list, **kwargs)
+
+        business_line_options = None
+        if self.is_user_adg_admin:
+            business_line_options = [DEFAULT_BUSINESS_LINE_FILTER, *self.business_line_titles]
+
+        context.update({
+            'application_type': self.application_type,
+            'total_applications': self.total_applications,
+            'business_lines': business_line_options,
+            'filter_by_business_line': self.filter_by_business_line,
+            'query': self.query,
+            'omni_courses_statuses': self.omni_courses_query_values,
+            'bu_courses_statuses': self.bu_courses_query_values,
+            'business_line_query': self.business_line_query,
+            'paginate_by': self.paginate_by,
+        })
+
+        return context
+
+    def filter_and_sort_applications(self, applications):
+        """
+        Filter the applications using the query params and sort the applications based
+        on date and course statuses
+
+        Returns:
+            QuerySet: Filtered and sorted queryset
+        """
+        if self.query:
+            applications = applications.filter(Q(user__profile__name__icontains=self.query) | Q(user__email=self.query))
+
+        if self.omni_courses_query_values:
+            applications = applications.filter(
+                user__application_hub__program_prereq_courses_status__in=self.omni_courses_query_values
+            )
+
+        if self.bu_courses_query_values:
+            applications = applications.filter(
+                user__application_hub__bu_prereq_courses_status__in=self.bu_courses_query_values
+            )
+
+        if self.is_user_adg_admin:
+            all_business_units_selected = self.filter_by_business_line == DEFAULT_BUSINESS_LINE_FILTER
+            if self.filter_by_business_line and not all_business_units_selected:
+                applications = applications.filter(
+                    business_line__title=self.filter_by_business_line
+                ).select_related('business_line')
+
+            if self.business_line_query and all_business_units_selected:
+                applications = applications.filter(
+                    business_line__title__in=self.business_line_query
+                ).select_related('business_line')
+
+        applications = self.sort_applications(applications)
```

Comment:
filter_and_sort_applications is mainly filtering the applications. It will be good if we can remove this call from here and place it in get_queryset method then we'll be complying with single-responsibility principle.